### PR TITLE
Decouple dojo toolkit code

### PR DESCRIPTION
**Pull request by lamenezes**
This PR decouple the code that was all concentrated in `main.py`

This will simplify the improvement  of the toolkit. It will make easier to implement:
- Test runners (for other python tests frameworks (pytest, nose, unittest) and for other languages :smile:) 
- Multi-platform notifications
- Improved timers

### CODE REVIEW

**Comment by lamenezes** [comment]
@zsinx6 @douglasshimada review :coffee:

**Comment by douglasshimada** [comment]
` class Dojo:
 14     def __init__(self, code_path, round_time=5, notifier=None, test_runner=None):
 15         print('Watching: {}\nTo change, reopen with path in first argument'.format(code_path))
 16 
 17         self.round_time = 5
`
line 17: why not self.round_time = round_time?

**Comment by douglasshimada** [comment]
git1
`self.round_time = round_time`

**Comment by douglasshimada** [comment]
Don't know if i'm trying to use it the wrong way. But when i ran python dojo.py i got some errorsgit2

**Comment by lamenezes** [comment]
@douglasshimada bem notado :smile: 

vc pode comentar em cima do código no github: https://github.com/grupy-sanca/dojo-toolkit/pull/12/files#diff-a2f5db558893bbef75748448ba1f009cR17

File: dojo_toolkit/dojo.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+from .notifier import GnomeNotifier
+from .test_runner import DoctestTestRunner
+from .timer import dojo_timer
+
+
+class Dojo:
+    def __init__(self, code_path, round_time=5, notifier=None, test_runner=None):
+        print('Watching: {}\nTo change, reopen with path in first argument'.format(code_path))
+
+        self.round_time = 5
```
Comment (lamenezes) [inline dojo_toolkit/dojo.py]: eita nois

**Comment by douglasshimada** [comment]
I remove "." in the imports
Original:
git3
 Mine:
git4
Then when i ran it worked
git6

**Comment by lamenezes** [comment]
solved

File: dojo_toolkit/test_runner.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+from subprocess import Popen
+
+
+class BaseTestRunner():
+    def __init__(self, code_path):
+        self.code_path = code_path
+
+
+class DoctestTestRunner(BaseTestRunner):
+    def run(self):
```
Comment (douglasshimada) [inline dojo_toolkit/test_runner.py]: Acho que se jogar o run para a BaseTestRunner() fica melhor.
`from subprocess import Popen


class BaseTestRunner():
    cmd = ""
    def __init__(self, code_path):
        self.code_path = code_path

    def run(self):
        self.cmd = self.cmd.format(self.code_path)
        process = Popen([cmd], shell=True)
        process.wait()
        return process.returncode == 0

class DoctestTestRunner(BaseTestRunner):
    cmd = "python -m doctest {}/*.py"
`

Comment (douglasshimada) [inline dojo_toolkit/test_runner.py]: Não consegui fazer o git mostrar bonito. xD
Fica assim:
test
Parece mais facil para add os testes depois.

File: dojo_toolkit/timer.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+def dojo_timer(notifier, round_time):
+    """Wait the defined and then shows notification and waits
+    for replace
+    """
+    while True:
+        notifier.notify('Time Up', timeout=15 * 1000)
+        print("\033c")
+        print('Press Enter when replaced')
+        raw_input()
```
Comment (douglasshimada) [inline dojo_toolkit/timer.py]: raw_input() vai dar erro com py3

**Comment by douglasshimada** [comment]
Aqui os testes não estão passando. =/
fail
fail2

**Comment by douglasshimada** [comment]
Os testes não estão passando. 
Fiz este teste:
"""
>>> True
True
"""
E falhou.

File: dojo_toolkit/test_runner.py (hunk 2 above)
Comment (lamenezes) [inline dojo_toolkit/test_runner.py]: A ideia é permitir rodar os testes sem ter que usar subprocess e usar o próprio módulo de  testes como por exemplo https://docs.python.org/3/library/doctest.html#basic-api

**Comment by lamenezes** [comment]
@douglasshimada os testes estão rodando aqui, vc apontou o toolkit para a pasta correta?

**Comment by douglasshimada** [comment]
Agora foi.
Tava testando na propria pasta do toolkit, dae deu erro.

File: dojo_toolkit/test_runner.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+        cmd = "python -m doctest {}/*.py".format(self.code_path)
```
Comment (zsinx6) [inline dojo_toolkit/test_runner.py]: Before showing the things, for minimal confusion is always good to clean the terminal and show **only** the new errors or test pass message

File: dojo_toolkit/dojo.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+from watchdog.observers import Observer
+
+from dojo_toolkit.code_handler import DojoCodeHandler
+from dojo_toolkit.notifier import GnomeNotifier
+from dojo_toolkit.test_runner import DoctestTestRunner
+from dojo_toolkit.timer import dojo_timer
+
+
+class Dojo:
```
Comment (zsinx6) [inline dojo_toolkit/dojo.py]: not a single line of documentation in this file

File: dojo_toolkit/notifier.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
+import os
+
+from pgi.repository import Notify, GdkPixbuf
+
+from .settings import ASSETS_DIR
+
+
+class BaseNotifier(object):
```
Comment (zsinx6) [inline dojo_toolkit/notifier.py]: documentation+1

File: dojo_toolkit/test_runner.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
+from subprocess import Popen
+
+
+class BaseTestRunner():
```
Comment (zsinx6) [inline dojo_toolkit/test_runner.py]: documentation+2

File: dojo_toolkit/test_runner.py (hunk 4 above)
Comment (lamenezes) [inline dojo_toolkit/test_runner.py]: I'm trying to apply the [single responsability principle](https://en.wikipedia.org/wiki/Single_responsibility_principle) to the project. Given that the TestRunner should be responsible to only run the tests.

We should have a better mechanism to control message display. The problem in your comment is being currently done by `clock_timer` (what an awful place for that to be). I think we can solve this issue display issue in another PR

File: dojo_toolkit/dojo.py (hunk 5 above)
Comment (lamenezes) [inline dojo_toolkit/dojo.py]: I believe building API docs should go to another PR, specially because this is only the first round of refactor that will happen to this project. 

Unless there is code that is too hard to understand (which should be rewritten).
